### ISSUE TITLE
[java] UseArraysAsList must ignore primitive arrays

### ISSUE THREAD

**Issue by huxi** [pmd/pmd#3379]
**Affects PMD Version:** 
6.36.0

**Rule:**
[UseArraysAsList](https://pmd.github.io/pmd-6.36.0/pmd_rules_java_performance.html#usearraysaslist)

**Description:**
Arrays of primitive data types must not trigger this rule.
This is especially problematic since replacing the code as suggested would introduce a bug, i.e. an Array with a single entry instead of an Array of values converted from primitive type to Object type (e.g. `short` into `Short`).

The following arrays should not trigger the rule:

- `byte[].class`
- `short[].class`
- `int[].class`
- `long[].class`
- `float[].class`
- `double[].class`
- `boolean[].class`
- `char[].class`

(I hope I didn't miss any.)

**Code Sample demonstrating the issue:**

```java
short[] array = (short[]) value;
ArrayList<Short> arrayList = new ArrayList<>(array.length);
for(short v:array) {
    arrayList.add(v);
}

// can't use this:
var list2 = Arrays.asList(array); // type of list2: List<short[]> and not List<Short>
```

**Expected outcome:**

PMD reports a violation at line `arrayList.add(v);` but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*
